src/prefect/logging/handlers.py

```diff
@@ -285,8 +286,13 @@ def handleError(self, record: logging.LogRecord) -> None:
         # Warn when a logger is used outside of a run context, the stack level here
         # gets us to the user logging call
         if isinstance(exc, MissingContextError):
-            warnings.warn(exc, stacklevel=8)
-            return
+            if PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW.value() == "warn":
```

Comment:
Retrieving the value has a non-negligible overhead and since logging calls are common we probably want to retrieve the value once rather than twice here.